libr/debug/p/native/w32.c

```diff
@@ -442,81 +429,70 @@ static int debug_exception_event (DEBUG_EVENT *de) {
 }
 
 static char *get_file_name_from_handle (HANDLE handle_file) {
-	HANDLE handle_file_map;
-	CHAR *filename = NULL;
-
+	HANDLE handle_file_map = NULL;
+	LPTSTR filename = NULL;
 	DWORD file_size_high = 0;
+	LPVOID map = NULL;
 	DWORD file_size_low = GetFileSize (handle_file, &file_size_high);
+
 	if (file_size_low == 0 && file_size_high == 0) {
 		return NULL;
 	}
-
-	handle_file_map = CreateFileMappingA (handle_file, NULL, PAGE_READONLY, 0, 1, NULL);
-
+	handle_file_map = CreateFileMapping (handle_file, NULL, PAGE_READONLY, 0, 1, NULL);
 	if (!handle_file_map) {
-		return NULL;
+		goto err_get_file_name_from_handle;
 	}
-	filename = malloc(MAX_PATH+1);
-
-	/* Create a file mapping to get the file name. */
-	void* map = MapViewOfFile (handle_file_map, FILE_MAP_READ, 0, 0, 1);
-
-	if (!map) {
-		free (filename);
-		CloseHandle (handle_file_map);
-		return NULL;
+	filename = malloc ((MAX_PATH + 1) * sizeof (TCHAR));
```

Comment:
Use wchar_t explicitly, you never want ANSI anyway.

Reply:
Comply with both windows calling convention and data types. UNICODE is already defined in file mk/platform.mk, TCHAR is wchar_t.